src/main/typescript/data/item/magic/properties.ts

```diff
@@ -0,0 +1,26 @@
+import { MagicType, getMagicType } from "../../../constants.js";
+import type WvItem from "../../../item/wvItem.js";
+import BaseItemProperties from "../common/baseItem/properties.js";
+import RulesProperties from "../common/rules/properties.js";
+import type MagicDataSource from "./source.js";
+import { MagicDataSourceData } from "./source.js";
+
+export default interface MagicDataProperties extends MagicDataSource {
+  data: MagicDataPropertiesData;
+}
+
+export class MagicDataPropertiesData
+  extends MagicDataSourceData
+  implements BaseItemProperties
+{
+  constructor(source: MagicDataSourceData, owningItem: WvItem) {
+    super();
+    foundry.utils.mergeObject(this, source);
+    this.type = getMagicType(this.school);
```

Comment:
You don't need to write that again in the constructor, if you already gave the property a default value.

Reply:
I did try that, but then it was just always set to unicorn magic, since the default value uses "general" to determine the type.